packages/http-client-csharp/emitter/src/lib/operation-converter.ts

```diff
@@ -207,6 +207,7 @@ function fromSdkHttpOperationParameter(
     Kind: getParameterKind(p, parameterType, rootApiVersions.length > 0),
     DefaultValue: getParameterDefaultValue(p.clientDefaultValue, parameterType),
     Decorators: p.decorators,
+    SkipUrlEncoding: (p as SdkPathParameter).allowReserved === true
```

Comment:
Type assertion is removed at runtime. If in the future other parameter types contain `allowReserved` property but require separate handling, the logic will be wrong.
```suggestion
    SkipUrlEncoding: (p.kind === "path" ? (p as SdkPathParameter).allowReserved : false)
```